### PR TITLE
Need Help - Do Not Merge

### PR DESCRIPTION
**Pull request by tuckermullen**
So I've successfully displayed the glyphicon boxes next to the items but I when I click the delete glyphicon, it gives me an error message `The action 'destroy' could not be found for UsersController` which seems simple enough to fix except I don't believe there should be a destroy action in the user controller at all. What am I doing wrong? This should connect to the destroy action in the items controller (which I've defined) but for some reason it isn't. Thanks! @bjblock

### CODE REVIEW

File: app/views/items/_item.html.erb [hunk 1]
```diff
@@ -1,5 +1,6 @@
 <% @items.each do |item| %>
   <ul>
-    <li><%= item.name %></li>
+    <li><%= link_to "", @item, method: :delete, class: 'gylphicon glyphicon-ok' %>
```
Comment (bjblock) [inline app/views/items/_item.html.erb]: this is the link you're having trouble with, right?
the `link_to` method takes 2 arguments (and then a hash of other optional arguments):
1. the text shown for the link (in this case there is no text and just the icon displayed by the class)
2. the url where the link should go to

It's argument 2 that is causing the problem.  You're using a rails shorthand which is to pass in an object that rails will then interpret into a link (e.g. if you pass in an item object with id of 5, rails will figure out that the url should be "/items/5").  You're passing `@item`.  What's the value of `@item`?

**Comment by tuckermullen** [comment]
So, by using rake routes, I can see that the `items#destroy` action requires a URL of `/users/:user_id/items/:id(.:format)`. With that behind said, wouldn't the code to build that be `[@user, @item.id]`? I also tried just calling a specific path but I keep getting the undefined method on any path I attempt to use. I've tried many different variations here, some break and throw an error and some show the correct template with the glyphicons but the link to the glyphicon itself is `localhost:3000/users/1` which obviously isn't right.

**Comment by bjblock** [comment]
You're asking some good questions and certainly may need to account for the nested route.  But before you get too far down that path, ask yourself what `@item` is.  Where have you defined that variable?  What's its value?  Can you do something like `<%= @item %>` to see what its value is?  It should be an item object with an id (so that rails can help build the url).  Is that what it is or something else?

**Comment by tuckermullen** [comment]
In the items_controller, I've defined `@item` as `@item = current_user.items.build(item_params)` which isn't exactly what I want here. I think I'm needing to find the item by id, `@find_item = Item.find(params[:item_id]`. So I created a `@find_item` instance variable under my create action in my items_controller and attempted to use it here as the 2nd argument in the link_to line but it still resulted in the same URL result of `localhost:3000/users/1`. Am I on the right track of trying to find the item by id? I feel like I'm making this harder than it needs to be by creating new instance variable to find the item id.

File: app/views/items/_item.html.erb (hunk 1 above)
Comment (bjblock) [inline app/views/items/_item.html.erb]: ok, let me come back to the code here.  It looks to me like you're looping through some collection of items called `@items`.  At least that's what I'm assuming is happening without looking at anything other than the few lines of code shown here.
So I'm assuming you're wanting to display a link for each item in the collection.  What variable do you use to represent each item in the collection?

Comment (tuckermullen) [inline app/views/items/_item.html.erb]: The variable I'm using to represent each item in the collection is `@items` which is set to: `@items = @user.items` in the controller. When I used `@items` as the 2nd argument, I get an error message: `undefined method to_model` within the browser.

I just looked at my controller (I may be wrong here), am I missing a defined variable that calls `Item.all`?

Comment (bjblock) [inline app/views/items/_item.html.erb]: the variable is `item`, not `@item`

**Pull request closed without merging**